src/id.rs

```diff
@@ -53,10 +56,11 @@ impl Id {
 
     pub fn with_keys(sign_keys: (crypto::sign::PublicKey, crypto::sign::SecretKey),
                      encrypt_keys: (crypto::box_::PublicKey, crypto::box_::SecretKey))-> Id {
+        let name = NameType::new(crypto::hash::sha512::hash(&sign_keys.0[..]).0); 
```

Comment:
Yes I think we will end up being more explicit here and using only the publicKey for non xor nodes.